### DISCUSSION TITLE
request help: Small probability running error

### DISCUSSION THREAD

**Discussion by whioue** [apache/apisix#5273] [Show and tell]
### Issue description

我尝试自己开发了一个插件，运行在access阶段，配置好路由并开启此插件后，当同时触发调用几十次此路由（每次调用此路由时传递给后置接口的参数不同）会有少许几次请求报500 Internal Server Error错误，其他请求可正常执行插件并得到后置接口的响应，我尝试查看日志发现有报错，具体如下：
1.插件开发示例：
![image](https://user-images.githubusercontent.com/47098518/137695108-b67cec34-111c-49d3-9a77-673065a42423.png)
2.错误日志：
![image](https://user-images.githubusercontent.com/47098518/137695182-0891829a-a3f5-4335-bfa7-689834b2e326.png)
3.抛出问题处定位：
![image](https://user-images.githubusercontent.com/47098518/137695257-9b4ff8e3-5bb7-4a62-8c63-feaf93ca84ad.png)

### Environment

- apisix version (cmd: `apisix version`): v2.8
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by tzssangglass** [comment]
1. This is code you wrote yourself, so I'll turn this issue into a discussion.
2. The information you provided is too little for me to reproduce.

**Reply by whioue** [reply]
> 1. 这是你自己写的代码，所以我会把这个问题变成一个讨论。
> 2. 您提供的信息太少，我无法重现。

此问题不容易重现，而且在错误日志中并没有看到我所写的插件中报错的内容，而是直接定位到apisix中init.lua中的报错，具体可以看上面的错误日志，所以我想问一下，这可能是什么原因导致的？

**Reply by tzssangglass** [reply]
> 此问题不容易重现

It is difficult to deduce the cause of errors that cannot be consistently reproduced.

> 直接定位到apisix中init.lua中的报错

That doesn't mean that there is an error in init.lua.

In short, I need a minimalist reproducible example.
